src/main/java/org/takes/facets/fork/FkAgent.java

```diff
@@ -0,0 +1,104 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.facets.fork;
+
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import lombok.EqualsAndHashCode;
+import org.takes.Request;
+import org.takes.Response;
+import org.takes.Take;
+import org.takes.facets.fork.am.AgentMatch;
+import org.takes.misc.Opt;
+import org.takes.rq.RqHeaders;
+
+/**
+ * Fork by user agent criteria accepted by "User-Agent" HTTP header.
+ *
+ * <p>The class is immutable and thread-safe.
+ *
+ * @author Valeriy Zhirnov (neonailol@gmail.com)
+ * @version $Id$
+ * @since 1.7.2
+ */
+@EqualsAndHashCode
+public final class FkAgent implements Fork {
+
+    /**
+     * Take to handle the request and dynamically return the response.
+     */
+    private final Take take;
+
+    /**
+     * User agent matcher.
+     */
+    private final AgentMatch match;
+
+    /**
+     * Ctor.
+     * @param take Take to handle the request dynamically.
+     * @param match Matcher.
+     */
+    public FkAgent(final Take take, final AgentMatch match) {
+        this.take = take;
+        this.match = match;
+    }
+
+    @Override
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+    public Opt<Response> route(final Request req) throws IOException {
+        final List<String> tokens = FkAgent.tokens(req);
+        Opt<Response> resp = new Opt.Empty<>();
+        for (final String token : tokens) {
+            if (this.match.matches(token)) {
+                resp = new Opt.Single<>(this.take.act(req));
+                break;
+            }
+        }
+        return resp;
+    }
+
+    /**
+     * Extract tokens from request.
+     * @param req Request.
+     * @return List of user agent tokens.
+     * @throws IOException If some problems inside.
+     */
+    private static List<String> tokens(final Request req) throws IOException {
+        final Pattern pattern = Pattern.compile("(\\w+)\\/([\\d\\.]+)");
```

Comment:
@neonailol I think it's better to make this pattern  `private static final`, because it produces immutable `Pattern` object (like a constant) and it will degrade performance if it called too often from outside.